coordinator/src/main/scala/filodb.coordinator/queryplanner/LogicalPlanUtils.scala

```diff
@@ -566,8 +565,30 @@ object LogicalPlanUtils extends StrictLogging {
       case sc: ScalarPlan => Some(Set.empty)  // since "None" will prevent optimization for all parent plans
       case _ => None
     }
-    helper(lp)
+    // Require that all raw-series selectors are configured with the same target-schema columns.
+    if (sameRawSeriesTargetSchemaColumns(lp, targetSchemaProvider, getShardKeyFilters).isDefined) helper(lp) else None
   }
   // scalastyle:on method.length
   // scalastyle:on cyclomatic.complexity
+
+  /**
+   * Returns true iff the entire plan tree preserves the argument label set.
+   * In other words, no aggregation "aggregates away" any of the argument labels.
+   */
+  def treePreservesLabels(lp: LogicalPlan, labels: Seq[String]): Boolean = lp match {
```

Comment:
should we consider a targeted unit test ( to test both pattern and anti-pattern lp  ), it might help us in future updates of this function

Reply:
Added. Excellent call btw -- this led me to add a new case:
```
case _: ScalarVaryingDoublePlan => false
```